doc/classes/Array.xml

```diff
@@ -619,6 +635,16 @@
 				func is_length_greater(a, b):
 				    return a.length() &gt; b.length()
 				[/codeblock]
+				This method can also be used to count how many elements in an array satisfy a certain condition, similar to [method count]:
+				[codeblock]
+				func is_even(val):
+				    return val % 2 == 0
+
+				func _ready():
+				    var arr = [1, 2, 3, 4, 5]
+				    var even_count = arr.reduce(func(count, next): return count + 1 if is_even(next) else count, 0) # Increment count if it's even, else leaves count the same.
```

Comment:
The line is very, very long. I would suggest moving the comment above, and potentially making it more meaningful. Code examples should speak for themselves, too.
```suggestion
				   # Increments count if the number is even. Otherwise, leaves count unchanged.
				    var even_count = arr.reduce(func(count, number): return count + 1 if is_even(number) else count, 0)
```

Also, I'm not exactly sure about this right now, but I believe you can omit `0` here (like one of the snippets above), which would make the example less confusing.

```suggestion
				   # Increments count if the number is even. Otherwise, leaves count unchanged.
				    var even_count = arr.reduce(func(count, number): return count + 1 if is_even(number) else count)
```

Reply:
The default argument is `nil`, so wouldn't `nil + 1` cause an error?

Reply:
I have added it anyway, though

Reply:
@AThousandShips What does the thumbs up mean in this case

Reply:
Agreement, that it will be called with `null`

Reply:
It won't cause a bug with `null`, but it *will* use the first element of the array as `count` instead of passing `0` and the first element

Reply:
Okay, yeah, that's an issue.